arlo_server/audit_boards.py

```diff
@@ -187,3 +190,101 @@ def list_audit_boards(
         serialize_audit_board(ab, round_status[ab.id]) for ab in audit_boards
     ]
     return jsonify({"auditBoards": json_audit_boards})
+
+
+def calculate_risk_measurements(election: Election, round: Round):
+    results = sample_results(election)
+
+    for contest in election.contests:
+        risk, is_complete = bravo.compute_risk(
+            election.risk_limit / 100,
+            sampler_contest.from_db_contest(contest),
+            results[contest.id],
+        )
+
+        round_contest = next(
+            rc for rc in round.round_contests if rc.contest_id == contest.id
+        )
+        round_contest.end_p_value = max(risk.values())
+        round_contest.is_complete = is_complete
```

Comment:
@umbernhard can you double check this?

Reply:
So the only thing about this is that we don't want to keep auditing contests that have met their risk limits. Is there logic somewhere else that keeps track of that?

Reply:
I copied this code from the single-page flow.

`is_complete` here signifies whether the contest met its risk limit this round, so yes, we do keep track of this.

Currently there's only one targeted contest that determines which ballots get sampled, so I don't think there's any logic to only audit contests that have not met their risk limits. We just keep auditing the ballots for the target contest until it meets its risk limit. Do we need to not show opportunistic contests that have met their risk limit to the audit boards when auditing ballots for the targeted contest?

Reply:
I think you're talking about in the Audit Board workflow @jonahkagan - do we still include those contests on the page the audit boards fill out for each ballot? If I'm understanding your q correctly, then we don't need to exclude opportunistic contests - if we have to audit another ballot for the target contest anyway, we might as well keep gathering data for the opportunistic contests (if they appear on that particular ballot, otherwise they're left blank).

Reply:
I dunno if that's a good idea. Because the test statistic for ballot polling is so ill-behaved, if we keep collecting data for opportunistic contests there's a chance that it could meet the risk-limit and then fall back above it once we have finished the targeted contest (e.g. if we only sample 10 more ballots for the targeted contest after the opportunistic one hits the risk limit, and the opportunistic votes are all for the loser), so our end result would be that the opportunistic contest didn't meet the risk-limit.

Reply:
I think that's ok - for v.1 on the opportunistic contests I'm certainly comfortable with that. For future versions, since we're checking at the end of every round it's easy enough to check if a contest has *ever* met the risk limit (at the end of any round) for audits using offline data entry, and for online data entry we'll want to implement the logic for "if it hasn't reached the risk limit at the end of the current round, use the ballot-by-ballot accounting to figure out if it ever crossed that threshold in the interim" ... but none of that is necessary for v.1.

Reply:
Ok, so for now I'm going to leave things as they are.